modules/publicize/publicize.php

```diff
@@ -780,17 +780,15 @@ function add_post_type_support() {
 	 * Register the Publicize Gutenberg extension
 	 */
 	function register_gutenberg_extension() {
-		jetpack_register_plugin( 'publicize', array( 'callback' => array( $this, 'get_extension_availability' ) ) );
-	}
+		// TODO: The `gutenberg/available-extensions` endpoint currently doesn't accept a post ID,
```

Comment:
We have to fix this - it currently introduces a regression.

Reply:
I don't think it does, TBH. Keep in mind that in case of Publicize, this is only used for the availability endpoint, so this will only be evaluated when that endpoint is it. (Gutenberg itself doesn't even have a concept of server-side plugin registration -- while there's `register_block_type`, there's no `register_plugin`).

However, that [endpoint](https://github.com/Automattic/jetpack/blob/91d89bc2033bf42fd6993ac712878866f46d4a5b/_inc/lib/core-api/wpcom-endpoints/gutenberg-available-extensions.php) doesn't currently support a `post` argument (and didn't before -- which is my fault, since I think Miguel was suggesting to add one). So [`_GET['post']`](https://github.com/Automattic/jetpack/pull/11091/files#diff-2f5f58835a08193f26adccabf4ae5ef3L787) was always empty, meaning that `current_user_can_access_publicize_data()` was always only called generically (checking the user's `publish_posts` capability in general, rather than for a given post ID specifically.